From 2a2e69a570b322e833ec227a5476d0399115c67e Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Mon, 05 May 2025 14:34:18 +0300
Subject: [PATCH] [SW-227941] serdes: separate oam data from rack data

Separate OAM data from Skyriver rack data to
different structures

Change-Id: I88ebdc286e4fc9439ad014db3854dafe1cb6a8a2
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/src/hl_serdes/gaudi3/serdes.c b/src/hl_serdes/gaudi3/serdes.c
index 431cc99..6ad8bef 100644
--- a/src/hl_serdes/gaudi3/serdes.c
+++ b/src/hl_serdes/gaudi3/serdes.c
@@ -48,9 +48,12 @@
 	uint64_t pol_tx_mask;
 };
 
-struct hl_serdes_rack_data {
+struct hl_serdes_oam_data {
 	uint8_t module_id;
 	uint8_t link_id;
+};
+
+struct hl_serdes_rack_data {
 	uint8_t node_id;
 	uint8_t num_of_nodes;
 	bool eeprom_opened;
@@ -399,6 +402,8 @@
 	.node_id = RACK_NODE_ID_UNDEFINED,
 };
 
+static struct hl_serdes_oam_data oam_data;
+
 #ifdef CONFIG_LFS_ENABLED
 static struct lfs_zephyrbd *zephyrbd_state;
 
@@ -501,16 +506,17 @@
 {
 	const struct hl_serdes_link_id_config *link_config;
 
-	rack_data.link_id = cpld_get_link_config();
-	rack_data.module_id = get_module_id();
-	rack_data.num_of_nodes = get_max_number_of_nodes(rack_data.link_id);
-	hl_serdes_config_init(rack_data.module_id, rack_data.link_id);
+	oam_data.link_id = cpld_get_link_config();
+	oam_data.module_id = get_module_id();
+
+	LOG_INF("Serdes: Module ID: %d Link ID: 0x%X", oam_data.module_id, oam_data.link_id);
+	rack_data.num_of_nodes = get_max_number_of_nodes(oam_data.link_id);
+	LOG_INF("Serdes: Max. number of nodes: %d", rack_data.num_of_nodes);
+	hl_serdes_config_init(oam_data.module_id, oam_data.link_id);
 	link_config = hl_serdes_link_id_config_get();
 	if (link_config->is_node_exists) {
 		cpld_get_node_id(&(rack_data.node_id));
-		LOG_INF("Module ID: %d Link ID: %d Node ID: %d Num. of nodes: %d",
-			rack_data.module_id, rack_data.link_id,
-			rack_data.node_id, rack_data.num_of_nodes);
+		LOG_INF("Serdes: Node ID: %d", rack_data.node_id);
 	}
 
 	return 0;
@@ -536,7 +542,7 @@
 
 	if (rack_data.eeprom_opened) {
 		rc = skyriver_get_taps_for_oam(rack_data.node_id,
-						rack_data.module_id, nic_info);
+						oam_data.module_id, nic_info);
 		if (rc)
 			LOG_ERR("Failed to read LAN taps from EEPROM: RC=%d", rc);
 	} else {
